### PR TITLE
feat(nuxt): add `fetchpriority` attribute and literal typings for meta components

### PR DESCRIPTION
**Pull request by sanjaiyan-dev**
Added fetchpriority attr in script and link component.

### CODE REVIEW

**Review by danielroe — APPROVED** [review]

**Comment by DamianGlowala** [comment]
Shall we narrow down the accepted values for the `fetchpriority` attribute, as per https://wicg.github.io/priority-hints/#definitions?

File: packages/nuxt/src/head/runtime/components.ts [hunk 1]
```diff
@@ -70,6 +70,7 @@ export const Script = defineComponent({
       default: undefined
     },
     defer: Boolean,
+    fetchpriority: String,
```
Comment (DamianGlowala) [inline packages/nuxt/src/head/runtime/components.ts:73]: https://wicg.github.io/priority-hints/#definitions
```
type FetchPriority = 'high' | 'low' | 'auto'
```

```suggestion
    fetchpriority: String as PropType<FetchPriority>,
```

File: packages/nuxt/src/head/runtime/components.ts [hunk 2]
```diff
@@ -118,6 +119,7 @@ export const Link = defineComponent({
     as: String,
     crossorigin: String,
     disabled: Boolean,
+    fetchpriority: String,
```
Comment (DamianGlowala) [inline packages/nuxt/src/head/runtime/components.ts:122]:
```suggestion
    fetchpriority: String as PropType<FetchPriority>,
```

**Review by DamianGlowala — CHANGES_REQUESTED** [review]

**Comment by DamianGlowala** [comment]
Great enhancement! I have closed #6262 in favor of your changes. There are two remarks that I've got:

- Why is boolean accepted for `crossorigin`? Some counterexamples:
    - https://github.com/posthtml/htmlnano/pull/79
    - https://github.com/nuxt/vue-meta/issues/447
- `x-ua-compatible` appears to be missing in `httpEquiv`, as per https://developer.mozilla.org/en-US/docs/Web/HTML/Element/meta#attr-http-equiv.

**Comment by DamianGlowala** [comment]
All I am thinking is whether the following would be preferable:

```
crossorigin: String as PropType<CrossOrigin>
```

Let's wait for others to express their thoughts, not sure which way to go :)

File: packages/nuxt/src/head/runtime/components.ts [hunk 3]
```diff
@@ -167,7 +168,7 @@ export const Base = defineComponent({
   props: {
     ...globalProps,
     href: String,
-    target: String,
+    target: String as PropType<Targets>,
```
Comment (DamianGlowala) [inline packages/nuxt/src/head/runtime/components.ts:171]: I think you meant `Target` (singular form) 👀

File: packages/nuxt/src/head/runtime/components.ts [hunk 4]
```diff
@@ -1,40 +1,56 @@
...
+import { useHead } from "./composables";
+import type {
+  Props,
+  FetchPriority,
+  CrossOrigin,
+  HTTPEquiv,
+  ReferrerPolicy,
+  LinkRelationship,
+  Target,
+} from "./types.ts";
```
Comment (danielroe) [inline packages/nuxt/src/head/runtime/components.ts:12]: Love the direction of this PR. Would you revert the various style/lint changes that have been made by your IDE?

**Review by danielroe — CHANGES_REQUESTED** [review]
Love the direction of this PR. Would you revert the various style/lint changes that have been made by your IDE?

**Comment by sanjaiyan-dev** [comment]
Made suggested change (not sure) 😄

File: packages/nuxt/src/head/runtime/types.ts [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+  | 'next'
+  | 'pingback'
+  | 'preconnect'
+  | 'prefetch'
+  | 'preload'
+  | 'prerender'
+  | 'prev'
+  | 'search'
+  | 'stylesheet'
+  | string
```
Comment (danielroe) [inline packages/nuxt/src/head/runtime/types.ts:45]: **A note** - no autocompletion will be provided for `LinkRelationship` and `Target` as providing `string` as a type effectively removes the string literal types.

**Review by danielroe — APPROVED** [review]
Really nice enhancement - thanks! ❤️

File: packages/nuxt/src/head/runtime/types.ts (hunk 5 above)
Comment (pi0) [reply packages/nuxt/src/head/runtime/types.ts:45]: Does it makes sense to keep the types then?

Comment (danielroe) [reply packages/nuxt/src/head/runtime/types.ts:45]: Yes, we could equally remove them without changing user experience.

Comment (danielroe) [reply packages/nuxt/src/head/runtime/types.ts]: Okay - small change keeps the autocompletion 👍

**Review by pi0 — APPROVED** [review]
Thanks!

**Comment by sanjaiyan-dev** [comment]
> Thanks!

My pleasure 💪🙌

File: packages/nuxt/src/head/runtime/types.ts (hunk 5 above)
Comment (DamianGlowala) [reply packages/nuxt/src/head/runtime/types.ts]: Nice one! An interesting approach to preserve autocompletion by treating the `(string & {})` as 'distinguishable' :) As a fun fact, it's also covered by the `LiteralUnion` type in the `type-fest` lib:

https://github.com/sindresorhus/type-fest/blob/main/source/literal-union.d.ts

**Pull request merged**